frontend/src/lib/components/apps/editor/component/components.ts

```diff
@@ -1686,6 +1688,60 @@ Hello \${ctx.username}
 			}
 		}
 	},
+	modalcomponent: {
+		name: 'Modal',
+		icon: SidebarClose,
+		dims: '1:1-2:1' as AppComponentDimensions,
+		customCss: {
+			button: { class: '', style: '' },
+			popup: { class: '', style: '' }
+		},
+		initialData: {
+			horizontalAlignment: 'center',
+			verticalAlignment: 'center',
+			configuration: {
+				modalTitle: {
+					type: 'static',
+					fieldType: 'text',
+					value: 'Modal title',
+					onlyStatic: true
+				},
+				label: {
+					type: 'static',
+					fieldType: 'text',
+					value: 'Press me'
+				},
+				color: {
+					fieldType: 'select',
+					type: 'static',
+					onlyStatic: true,
+					selectOptions: buttonColorOptions,
+					value: 'blue'
+				},
+				size: {
+					fieldType: 'select',
+					type: 'static',
+					onlyStatic: true,
+					selectOptions: selectOptions.buttonSizeOptions,
+					value: 'xs'
+				},
+				fillContainer: {
+					fieldType: 'boolean',
+					type: 'static',
+					onlyStatic: true,
+					value: false
+				},
+				disabled: {
+					fieldType: 'boolean',
+					type: 'static',
+					value: false
+				}
```

Comment:
we should add tooltip, also might be worth naming all those:
buttonSize, buttonFillContainer etc to avoid confusion